### PR TITLE
feat: dumbo compress helper

### PR DESCRIPTION
**Pull request by darkterminal**
Related to #37

### CODE REVIEW

File: src/Helpers/Compress.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    /**
+     * Perform the actual compression of the response body using the specified encoding
+     *
+     * @param string $body The response body to compress
+     * @param string $encoding The encoding to use for compression
+     *
+     * @return string|StreamInterface The compressed response body
+     */
+    private static function performCompression($body, string $encoding)
```
Comment (notrab) [inline src/Helpers/Compress.php]: Do we need a try/catch block here, a return type, and support for `br`?

For example:

```php
 try {
        $compressed = match ($encoding) {
            'gzip' => gzencode((string)$body),
            'deflate' => gzdeflate((string)$body),
            'br' => extension_loaded('brotli') ? brotli_compress((string)$body) : (string)$body,
            default => (string)$body,
        };
        return Utils::streamFor($compressed);
    } catch (\Exception $e) {
        // Log the error
        error_log("Compression failed: " . $e->getMessage());
        return Utils::streamFor((string)$body);
    }
```

**Review by notrab — COMMENTED** [review]
Looks great @darkterminal! I just have one question, the `performCompression` method.

File: src/Helpers/Compress.php (hunk 1 above)
Comment (darkterminal) [reply src/Helpers/Compress.php]: I never use `brotli` but it's look good. And also the try/catch, I think we don't need that cz the default fallback is string CMIIW.

Comment (notrab) [reply src/Helpers/Compress.php]: Sounds good!

In that case, we could do something like this:

```php
$content = (string) $body;
        return match ($encoding) {
            'br' => Utils::streamFor(self::brotliCompress($content)),
            'gzip' => Utils::streamFor(gzencode($content)),
            'deflate' => Utils::streamFor(gzdeflate($content)),
            default => $body,
        };
```

Which with the `brotliComrpess` method raising an error if it's not avalable:

```php
private static function brotliCompress(string $content): string|false
    {
        if (!function_exists('brotli_compress')) {
            throw new \RuntimeException('Brotli compression is not available. Make sure the Brotli PHP extension is installed and enabled.');
        }
        return brotli_compress($content);
    }
```

File: src/Helpers/Compress.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                $response->hasHeader('Content-Encoding') ||
+                $ctx->req->method() === 'HEAD' ||
+                ($contentLength && (int) $contentLength < $threshold) ||
+                !self::shouldCompress($response) ||
+                !self::shouldTransform($response)
+            ) {
+                return $response;
+            }
+
+            $acceptedEncodings = explode(',', $ctx->req->header('Accept-Encoding'));
```
Comment (notrab) [inline src/Helpers/Compress.php]:
```suggestion
            $acceptedEncodings = array_map('trim', explode(',', $ctx->req->header('Accept-Encoding')));
```

We should probably trim any whitespace here, just to be safe when used outside a browser. Using the array means we get to check the array, and not the value passed too. Some clients might send values as `gzip,br`, `br, gzip`, `deflate,gzip`, etc. etc.

File: src/Helpers/Compress.php (hunk 1 above)
Comment (darkterminal) [reply src/Helpers/Compress.php]: Wooo... you got it bro!!!! :dancers:

File: src/Helpers/Compress.php (hunk 2 above)
Comment (darkterminal) [reply src/Helpers/Compress.php]: Make sense!

**Comment by darkterminal** [comment]
The song entitled Compress has finished playing.

**Comment by notrab** [comment]
@darkterminal I tested Brotli locally, but ran into some issues. I've reverted those changes, but added a README and updated example for now.

Let's add Brotli soon!

**Pull request merged**